legacy/python/platform/src/actions/install_deps.rs

```diff
@@ -0,0 +1,80 @@
+use moon_action::Operation;
+use moon_console::{Checkpoint, Console};
+use moon_python_tool::PythonTool;
+use std::path::Path;
+
+use crate::find_requirements_txt;
+
+pub async fn install_deps(
+    python: &PythonTool,
+    workspace_root: &Path,
+    working_dir: &Path,
+    console: &Console,
+) -> miette::Result<Vec<Operation>> {
+    let mut operations = vec![];
+
+    if let Some(pip_config) = &python.config.pip {
+        let requirements_path = find_requirements_txt(working_dir, workspace_root);
+        let virtual_environment = if python.config.root_requirements_only {
+            &workspace_root.join(python.config.venv_name.clone())
+        } else {
+            &working_dir.join(python.config.venv_name.clone())
+        };
+
+        if !virtual_environment.exists() {
+            console
+                .out
+                .print_checkpoint(Checkpoint::Setup, "activate virtual environment")?;
+            let args = vec![
+                "-m",
+                "venv",
+                virtual_environment.as_os_str().to_str().unwrap(),
+            ];
+            operations.push(
+                Operation::task_execution(format!("python {} ", args.join(" ")))
+                    .track_async(|| python.exec_python(args, workspace_root))
+                    .await?,
+            );
+        }
+
+        if let Some(install_args) = &pip_config.install_args {
```

Comment:
This should probably still `pip install` even if there are no custom args yeah? The args from the config should just append.

Reply:
Yes you are right, even with the next comment (difference first and second pip install)
Therefore I have re-worked the strategy that in before the `args` will be constructed.

And if these are not empty, only one `pip install` will be executed.